repose-aggregator/commons/utilities/src/main/scala/org/openrepose/commons/utils/servlet/http/HttpServletResponseWrapper.scala

```diff
@@ -46,84 +53,253 @@ import javax.servlet.http.HttpServletResponse
-   * @param headerMode the mode to use for header accessiblity
+   * @param headerMode the mode to use for header accessibility
    * @param bodyMode the mode to use for body accessibility
    */
   def this(originalResponse: HttpServletResponse, headerMode: ResponseMode, bodyMode: ResponseMode) =
     this(originalResponse, headerMode, bodyMode, originalResponse.getOutputStream)
 
-  override def getHeaderNamesList: util.List[String] = ???
+  override def getHeaderNamesList: util.List[String] = getHeaderNames.toList
+
+  override def getHeaderNames: util.Collection[String] = headerMap.keys
+
+  override def containsHeader(name: String): Boolean = headerMap.contains(name)
+
+  override def getHeader(name: String): String = getHeaderValues(name).headOption.orNull
+
+  override def getHeaders(name: String): util.Collection[String] = getHeaderValues(name)
+
+  override def getHeadersList(name: String): util.List[String] = getHeaderValues(name)
+
+  override def getPreferredHeaders(name: String): util.List[String] = getPreferredHeaderValues(getHeaderValues(name))
+
+  override def getPreferredSplittableHeaders(name: String): util.List[String] =
+    getPreferredHeaderValues(getSplittableHeaders(name))
+
+  override def getSplittableHeaders(name: String): util.List[String] =
+    getHeaderValues(name).foldLeft(List.empty[String])((list, value) => list ++ value.split(","))
+
+  /**
+   * @throws IllegalStateException when headerMode is anything other than ResponseMode.MUTABLE
+   */
+  override def addHeader(name: String, value: String): Unit = {
+    ifMutable(headerMode) {
+      headerMap = headerMap + (name -> (headerMap.getOrElse(name, Seq.empty[String]) :+ value))
+    }
+  }
+
+  /**
+   * @throws IllegalStateException when headerMode is anything other than ResponseMode.MUTABLE
+   */
+  override def addHeader(name: String, value: String, quality: Double): Unit =
+    addHeader(name, s"$value;q=$quality")
 
-  override def addHeader(headerName: String, headerValue: String, quality: Double): Unit = ???
+  /**
+   * @throws IllegalStateException when headerMode is anything other than ResponseMode.MUTABLE
+   */
+  override def addIntHeader(name: String, value: Int): Unit = addHeader(name, value.toString)
 
   /**
+   * Formats the input time (as milliseconds since the epoch) to a format defined in RFC2616.
+   *
    * @throws IllegalStateException when headerMode is anything other than ResponseMode.MUTABLE
    */
-  override def appendHeader(headerName: String, headerValue: String): Unit = ???
+  override def addDateHeader(name: String, timeSinceEpoch: Long): Unit =
+    addHeader(name, DateUtils.formatDate(new Date(timeSinceEpoch)))
 
   /**
    * @throws IllegalStateException when headerMode is anything other than ResponseMode.MUTABLE
    */
-  override def appendHeader(headerName: String, headerValue: String, quality: Double): Unit = ???
+  override def appendHeader(name: String, value: String): Unit = {
```

Comment:
this looks like it would succeed regardless of mode if there was already a value.